src/main/scala/zalando/meetup/bracket/DatabaseOperations.scala

```diff
@@ -25,10 +24,20 @@ object DatabaseOperations {
   def createDataSource[F[_] : Sync](conf: DatabaseConfig): Resource[F, HikariDataSource] =
     Resource.fromAutoCloseable(createDSConfig[F](conf).map(new HikariDataSource(_)))
 
+  def connectionResource[F[_]: Sync](ds: DataSource): Resource[F, Connection] = Resource.fromAutoCloseable(Sync[F].catchNonFatal(ds.getConnection()))
```

Comment:
Why are you using here `catchNonFatal` but `delay` below?

Reply:
Because I copy pasted one and I wrote the other one :), I don't think it matters here tbh. Which one do you like the most?

Reply:
:thinking: looking at the semantics, `catchNonFatal` is pure, so memoised, in this case it is the same, but I would rather use `delay`

Reply:
👍